Src/VsSpecific/VsSpecific/CSharpScriptExecutor.cs

```diff
@@ -0,0 +1,172 @@
+﻿#if VS_SPECIFIC_2017 || VS_SPECIFIC_2019
+
+using System;
+using Vim.Interpreter;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+namespace Vim.VisualStudio.Specific
+{
+    internal sealed class CSharpScriptExecutor : ICSharpScriptExecutor
+    {
+        private const string ScriptFolder = "vsvimscripts";
+        private Dictionary<string, object> _scripts = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+
+        private MethodInfo _cSharpScriptCreate = null;
+        private dynamic _scriptOptions = null;
+        private dynamic _defaultScriptSourceResolver = null;
+        private dynamic _defaultScriptMetadataResolver = null;
+        private dynamic _defaultScriptOptions = null;
+
+        private void Execute(IVim vim, CallInfo callInfo, bool createEachTime)
+        {
+            try
+            {
+                string assemblyPath;
+                string baseDirectory = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"CommonExtensions\Microsoft\ManagedLanguages\VBCSharp\InteractiveComponents");
+
+                string assemblyName;
+                if (_cSharpScriptCreate == null)
+                {
+                    assemblyName = "Microsoft.CodeAnalysis.CSharp.Scripting.dll";
+                    assemblyPath = Path.Combine(baseDirectory, assemblyName);
+                    if (!File.Exists(assemblyPath))
+                    {
+                        vim.ActiveStatusUtil.OnError($"{assemblyName} not found.");
+                        return;
+                    }
+                    var asm = Assembly.LoadFile(assemblyPath);

```

Comment:
Rather than looking for this on disk have you tried just using `Assembly.Load(AssemblyName)`? That is the "better" way to load assemblies (avoids load context issues). In this particular case the path should be on the probing path of Visual Studio so this should work I believe.